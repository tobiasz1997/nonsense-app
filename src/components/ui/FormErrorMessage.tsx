import { FC, PropsWithChildren } from 'react';

const FormErrorMessage: FC<PropsWithChildren> = (props) => {
	return <div className="my-2 text-xs text-red">{props.children}</div>;
};

export default FormErrorMessage;
