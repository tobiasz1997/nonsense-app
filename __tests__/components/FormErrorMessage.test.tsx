import { render, screen } from '@testing-library/react';
import FormErrorMessage from '@components/ui/FormErrorMessage';

describe('Form Error Message', () => {
	const content = '@#$%^&*()qwerty';

	it('should render without crashing', () => {
		const { container } = render(
			<FormErrorMessage>{content}</FormErrorMessage>
		);
		expect(container).toMatchSnapshot();
	});

	it('should message be correct', () => {
		render(<FormErrorMessage>{content}</FormErrorMessage>);

		const text = screen.getByText(content);

		expect(text).toBeInTheDocument();
	});
});
