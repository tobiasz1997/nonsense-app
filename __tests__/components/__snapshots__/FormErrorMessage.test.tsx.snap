// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Error Message should render without crashing 1`] = `
<div>
  <div
    class="my-2 text-xs text-red"
  >
    @#$%^&*()qwerty
  </div>
</div>
`;
